tls/s2n_post_handshake.h

```diff
@@ -15,7 +15,12 @@
 
 #pragma once
 
-#include "tls/s2n_connection.h"
+struct s2n_connection;
```

Comment:
Why would we do this?

Reply:
To avoid a circular dependency. I added the "s2n_post_handshake" structure to s2n_connection. s2n_connection therefore needs to know the size of that structure.

s2n_post_handshake.h only needs to know that s2n_connection exists, since it only uses pointers to connections in its methods.